tdb/vidrl_upload.py

```diff
@@ -28,116 +29,120 @@ def read_vidrl(path, fstem, assay_type):
+    serum_id_pattern = r"^[A-Z]\d{4,8}$"
+    serum_passage_pattern = r"(MDCK\d+|SIAT\d+|E\d+)"
+    serum_abbrev_pattern = r"\w+\s{0,1}\w+/\d+.*"
+    crick = False
+
+    for worksheet in workbook.sheets():
+        # autodetecting titer, strain, serum blocks
+        titer_block = find_titer_block(worksheet)
+
+        if len(titer_block["col_start"]) == 0:
+            print("No titer block found.")
+            break
+
+        titer_coords = {
+            'col_start': titer_block["col_start"][0][0],
+            'col_end': titer_block["col_end"][0][0],
+            'row_start': titer_block["row_start"][0][0],
+            'row_end': titer_block["row_end"][0][0]
+        }
+
+        virus_block = find_virus_columns(
+            worksheet=worksheet,
+            titer_coords=titer_coords,
+            virus_pattern=virus_pattern,
+            virus_passage_pattern=virus_passage_pattern,
+        )
+
+        serum_block = find_serum_rows(
+            worksheet=worksheet,
+            titer_coords=titer_coords,
+            virus_names=virus_block["virus_names"],
+            serum_id_pattern=serum_id_pattern,
+            serum_passage_pattern=serum_passage_pattern,
+            serum_abbrev_pattern=serum_abbrev_pattern,
+            crick=crick,
+        )
+
+        # Print the most likely row and column indices for the titer block
+        print("Titer block:")
+        print(f"  Most likely (n={titer_block['col_start'][0][1]}) col_start: {titer_block['col_start'][0][0]}")
+        print(f"  Most likely (n={titer_block['col_end'][0][1]}) col_end: {titer_block['col_end'][0][0]}")
+        print(f"  Most likely (n={titer_block['row_start'][0][1]}) row_start: {titer_block['row_start'][0][0]}")
+        print(f"  Most likely (n={titer_block['row_end'][0][1]}) row_end: {titer_block['row_end'][0][0]}")
+
+        # Print virus and serum annotations row and column indices
+        # Print Virus and Serum annotations row and column indices
+        print("Virus (antigen) block: left and right of the titer block")
+        print(f"  virus column index: {virus_block['virus_col_idx']}")
+        print(f"  virus passage column index: {virus_block['virus_passage_col_idx']}")
+        print(f"  virus names: {virus_block['virus_names']}")
+
+        print("Serum (antisera) block: above the titer block")
+        print(f"  serum ID row index: {serum_block['serum_id_row_idx']}")
+        print(f"  serum passage row index: {serum_block['serum_passage_row_idx']}")
+        print(f"  serum abbreviated name row index: {serum_block['serum_abbrev_row_idx']}")
+
+        # Match abbreviated names across the top to the full names along the left side and auto convert to full names
+        if serum_block["serum_abbrev_row_idx"] is not None:
+            print("serum_mapping = {")
+            for abbrev, full in serum_block["serum_mapping"].items():
+                print(f"    '{abbrev}': '{full}',")
+            print("}")
```

Comment:
Ohh, I like this suggestion but will hold off for now, mostly because the `print(f"    '{abbrev}': '{full}',")` is a reminder to myself that `abbrev` is the key, and `full` is the value.

I'll keep this suggestion in mind for the future.